test/ws_notification_test/ws_notification_test.go

```diff
@@ -235,52 +236,77 @@ var _ = Describe("WS", func() {
+
+		assertPlatformIsActive := func() {
+			obj, err := repository.Get(context.TODO(), types.PlatformType, idCriteria)
 			Expect(err).ShouldNot(HaveOccurred())
+			Expect(obj.(*types.Platform).Active).To(BeTrue())
+		}
 
-			idCriteria := query.Criterion{
+		BeforeEach(func() {
+			pongCh = make(chan struct{})
+			newPlatform = common.RegisterPlatformInSM(common.GenerateRandomPlatform(), ctx.SMWithOAuth, map[string]string{})
+			idCriteria = query.Criterion{
 				LeftOp:   "id",
 				Operator: query.EqualsOperator,
-				RightOp:  []string{platform.ID},
+				RightOp:  []string{newPlatform.ID},
 				Type:     query.FieldQuery,
 			}
-			obj, err := repository.Get(context.TODO(), types.PlatformType, idCriteria)
-			Expect(err).ShouldNot(HaveOccurred())
-			Expect(obj.(*types.Platform).Active).To(BeTrue())
-		})
-	})
+			Expect(newPlatform.Active).To(BeFalse())
 
-	Context("when platform disconnects", func() {
-		It("should switch platform's active status to false", func() {
-			conn, _, err := ctx.ConnectWebSocket(platform, queryParams)
+			var err error
+			conn, _, err = ctx.ConnectWebSocket(newPlatform, queryParams)
 			Expect(err).ShouldNot(HaveOccurred())
 
-			ctx.CloseWebSocket(conn)
+			conn.SetPongHandler(func(data string) error {
+				Expect(data).To(Equal("pingping"))
+				close(pongCh)
+				return nil
+			})
+
+			go func() {
+				_, _, err := conn.ReadMessage()
+				Expect(err).Should(HaveOccurred())
+			}()
+		})
 
-			idCriteria := query.Criterion{
-				LeftOp:   "id",
-				Operator: query.EqualsOperator,
-				RightOp:  []string{platform.ID},
-				Type:     query.FieldQuery,
-			}
-			ctx, _ := context.WithTimeout(context.TODO(), 5*time.Second)
-			ticker := time.NewTicker(500 * time.Millisecond)
-			for {
-				select {
-				case <-ticker.C:
-					obj, err := repository.Get(context.TODO(), types.PlatformType, idCriteria)
-					Expect(err).ShouldNot(HaveOccurred())
-					p := obj.(*types.Platform)
-					if p.Active == false && !p.LastActive.IsZero() {
-						return
+		Context("when ping is received", func() {
+			It("should switch platform's active status to true", func() {
+				Expect(conn.WriteControl(websocket.PingMessage, []byte("pingping"), time.Now().Add(pingTimeout))).ShouldNot(HaveOccurred())
+				Eventually(pongCh).Should(BeClosed()) // wait for a pong message
+				assertPlatformIsActive()
+			})
+		})
+
+		Context("when ping is not received", func() {
+			It("should switch platform's active status to false", func() {
+				Expect(conn.WriteControl(websocket.PingMessage, []byte("pingping"), time.Now().Add(pingTimeout))).ShouldNot(HaveOccurred())
+				Eventually(pongCh).Should(BeClosed()) // wait for a pong message
+				assertPlatformIsActive()
+
+				By(fmt.Sprintf("Ping received and active status is true, then when %v ping timeout passes, active status should be set to false", pingTimeout))
+
+				ctx, _ := context.WithTimeout(context.TODO(), pingTimeout+time.Second)
+				ticker := time.NewTicker(500 * time.Millisecond)
```

Comment:
Can we do this a fraction of the `pingTimeout`? Let's say pingTimeout/3 for example.
Minor: Also is it possible to reduce the pingTimeout to lets say 1/2 sec.? It might speed up the tests.

Reply:
It is 1 second currently the BeforeEach overrides it (https://github.com/Peripli/service-manager/pull/339/files/84a78c676dc2d747851c0ec5a7a9099a0bb672ff#diff-234cfb3ff7d03d66a6db4f491358a003R74)